libraries/AP_Compass/AP_Compass_SITL.cpp

```diff
@@ -123,19 +134,19 @@ void AP_Compass_SITL::_timer()
         correct_field(f, _compass_instance[i]);
 
         _mag_accum[i] += f;
+        _accum_count[i]++;
     }
 
     if (!_sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
         return;
     }
```

Comment:
I know this isn't part of your changes but just something that caught my eye...

If this call to take the semaphore will wait forever, what would make take() return false??